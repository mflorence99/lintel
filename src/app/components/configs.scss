:host {
  display: block;
  height: 100%;
}

.config {
  cursor: pointer;
  line-height: 150%;
  margin-bottom: 1rem;

  .category,
  .filename {
    color: var(--vscode-editorWidget-foreground);
    font-weight: normal;

    &.selected {
      color: var(--vscode-settings-headerForeground);
      font-weight: bold;
    }

  }

}
