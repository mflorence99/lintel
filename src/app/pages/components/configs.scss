:host {
  display: block;
  height: 100%;
  overflow-y: auto;
}

.config {
  cursor: pointer;
  line-height: 175%;
  margin-bottom: 1rem;

  .item {
    align-items: center;
    color: var(--vscode-editorWidget-foreground);
    display: grid;
    font-weight: normal;
    grid-template-columns: 1.5rem 1fr;

    > div {
      line-height: 150%;
    }

    > div:first-of-type {
      justify-self: center;
    }

    fa-icon {
      opacity: var(--lintel-opacity-icon);
    }

    &.selected {
      color: var(--vscode-settings-headerForeground);
      font-weight: bold;

      fa-icon {
        opacity: 1;
      }

    }

    &.selected.unknown {
      color: var(--vscode-inputValidation-errorBorder);
    }

  }

}

.logo {
  padding: 1rem;

  img {
    width: 100%;
  }

}
