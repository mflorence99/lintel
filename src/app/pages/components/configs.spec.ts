import { ConfigsComponent } from './configs';
import { TestBed } from '@angular/core/testing';

import { async } from '@angular/core/testing';
import { prepare } from './component.spec';

describe('ConfigsComponent', () => {

  beforeEach(async(() => prepare([ConfigsComponent])));

  test('Component is created', () => {
    const fixture = TestBed.createComponent(ConfigsComponent);
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });

  test('Category can be selected', () => {
    const fixture = TestBed.createComponent(ConfigsComponent);
    const component = fixture.componentInstance;
    component.selectCategory(new Event('click'), 'Best Practices');
    expect(component.selection.category).toEqual('Best Practices');
    // selecting category again is harmless
    component.selectCategory(new Event('click'), 'Best Practices');
    expect(component.selection.category).toEqual('Best Practices');
  });

  test('Filename can be selected', () => {
    const fixture = TestBed.createComponent(ConfigsComponent);
    const component = fixture.componentInstance;
    component.selectFileName(new Event('click'), 'package.json');
    expect(component.selection.fileName).toEqual('package.json');
  });

});
