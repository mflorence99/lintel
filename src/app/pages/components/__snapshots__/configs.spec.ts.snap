// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigsComponent snapshot 1`] = `
<lintel-configs
  __ngContext__={[Function LRootView]}
  actions$={[Function Actions]}
  cdf={[Function ViewRef]}
  configs={[Function ConfigsState]}
  destroy$={[Function DestroyService]}
  filter={[Function class_1]}
  lintel={[Function class_1]}
  params={[Function Params]}
  selection={[Function class_1]}
  utils={[Function Utils]}
>
  <header
    class="logo"
  >
    <img
      src="assets/lintel.png"
    />
  </header><menu
    class="config"
  >
    <li
      class="item selected"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="lg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-node-js fa-w-14 fa-fw fa-lg"
            data-icon="node-js"
            data-prefix="fab"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 508c-6.7 0-13.5-1.8-19.4-5.2l-61.7-36.5c-9.2-5.2-4.7-7-1.7-8 12.3-4.3 14.8-5.2 27.9-12.7 1.4-.8 3.2-.5 4.6.4l47.4 28.1c1.7 1 4.1 1 5.7 0l184.7-106.6c1.7-1 2.8-3 2.8-5V149.3c0-2.1-1.1-4-2.9-5.1L226.8 37.7c-1.7-1-4-1-5.7 0L36.6 144.3c-1.8 1-2.9 3-2.9 5.1v213.1c0 2 1.1 4 2.9 4.9l50.6 29.2c27.5 13.7 44.3-2.4 44.3-18.7V167.5c0-3 2.4-5.3 5.4-5.3h23.4c2.9 0 5.4 2.3 5.4 5.3V378c0 36.6-20 57.6-54.7 57.6-10.7 0-19.1 0-42.5-11.6l-48.4-27.9C8.1 389.2.7 376.3.7 362.4V149.3c0-13.8 7.4-26.8 19.4-33.7L204.6 9c11.7-6.6 27.2-6.6 38.8 0l184.7 106.7c12 6.9 19.4 19.8 19.4 33.7v213.1c0 13.8-7.4 26.7-19.4 33.7L243.4 502.8c-5.9 3.4-12.6 5.2-19.4 5.2zm149.1-210.1c0-39.9-27-50.5-83.7-58-57.4-7.6-63.2-11.5-63.2-24.9 0-11.1 4.9-25.9 47.4-25.9 37.9 0 51.9 8.2 57.7 33.8.5 2.4 2.7 4.2 5.2 4.2h24c1.5 0 2.9-.6 3.9-1.7s1.5-2.6 1.4-4.1c-3.7-44.1-33-64.6-92.2-64.6-52.7 0-84.1 22.2-84.1 59.5 0 40.4 31.3 51.6 81.8 56.6 60.5 5.9 65.2 14.8 65.2 26.7 0 20.6-16.6 29.4-55.5 29.4-48.9 0-59.6-12.3-63.2-36.6-.4-2.6-2.6-4.5-5.3-4.5h-23.9c-3 0-5.3 2.4-5.3 5.3 0 31.1 16.9 68.2 97.8 68.2 58.4-.1 92-23.2 92-63.4z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        lintel/â€‹package.json
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        ESLint Config
      </div>
    </li>
    <li
      class="item selected"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        *.*
      </div>
    </li>
    
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        *.md
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        *.html
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        *.ts, *.tsx
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        *.empty
      </div>
    </li>
    <li
      class="item inherited"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        *.cjs, .*.cjs
      </div>
    </li>
    <li
      class="item inherited"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        *.mjs, .*.mjs
      </div>
    </li>
    
    <li
      class="item selected"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-plug fa-w-12 fa-fw"
            data-icon="plug"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M320,32a32,32,0,0,0-64,0v96h64Zm48,128H16A16,16,0,0,0,0,176v32a16,16,0,0,0,16,16H32v32A160.07,160.07,0,0,0,160,412.8V512h64V412.8A160.07,160.07,0,0,0,352,256V224h16a16,16,0,0,0,16-16V176A16,16,0,0,0,368,160ZM128,32a32,32,0,0,0-64,0v96h64Z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        eslint
      </div>
    </li>
    <li
      class="item selected"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        Active Rules
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        Best Practices
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        ECMAScript 6
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        Node.js And CommonJS
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        Possible Errors
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        Strict Mode
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        Stylistic Issues
      </div>
    </li>
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="xs"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-fw fa-xs"
            data-icon="circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        Variables
      </div>
    </li>
    
    
    
  </menu><menu
    class="config"
  >
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="lg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-js fa-w-14 fa-fw fa-lg"
            data-icon="js"
            data-prefix="fab"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 32v448h448V32H0zm243.8 349.4c0 43.6-25.6 63.5-62.9 63.5-33.7 0-53.2-17.4-63.2-38.5l34.3-20.7c6.6 11.7 12.6 21.6 27.1 21.6 13.8 0 22.6-5.4 22.6-26.5V237.7h42.1v143.7zm99.6 63.5c-39.1 0-64.4-18.6-76.7-43l34.3-19.8c9 14.7 20.8 25.6 41.5 25.6 17.4 0 28.6-8.7 28.6-20.8 0-14.4-11.4-19.5-30.7-28l-10.5-4.5c-30.4-12.9-50.5-29.2-50.5-63.5 0-31.6 24.1-55.6 61.6-55.6 26.8 0 46 9.3 59.8 33.7L368 290c-7.2-12.9-15-18-27.1-18-12.3 0-20.1 7.8-20.1 18 0 12.6 7.8 17.7 25.9 25.6l10.5 4.5c35.8 15.3 55.9 31 55.9 66.2 0 37.8-29.8 58.6-69.7 58.6z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        el-3270/â€‹.eslintrc.js
      </div>
    </li>
    
  </menu><menu
    class="config"
  >
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="lg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-file-code fa-w-12 fa-fw fa-lg"
            data-icon="file-code"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M149.9 349.1l-.2-.2-32.8-28.9 32.8-28.9c3.6-3.2 4-8.8.8-12.4l-.2-.2-17.4-18.6c-3.4-3.6-9-3.7-12.4-.4l-57.7 54.1c-3.7 3.5-3.7 9.4 0 12.8l57.7 54.1c1.6 1.5 3.8 2.4 6 2.4 2.4 0 4.8-1 6.4-2.8l17.4-18.6c3.3-3.5 3.1-9.1-.4-12.4zm220-251.2L286 14C277 5 264.8-.1 252.1-.1H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48V131.9c0-12.7-5.1-25-14.1-34zM256 51.9l76.1 76.1H256zM336 464H48V48h160v104c0 13.3 10.7 24 24 24h104zM209.6 214c-4.7-1.4-9.5 1.3-10.9 6L144 408.1c-1.4 4.7 1.3 9.6 6 10.9l24.4 7.1c4.7 1.4 9.6-1.4 10.9-6L240 231.9c1.4-4.7-1.3-9.6-6-10.9zm24.5 76.9l.2.2 32.8 28.9-32.8 28.9c-3.6 3.2-4 8.8-.8 12.4l.2.2 17.4 18.6c3.3 3.5 8.9 3.7 12.4.4l57.7-54.1c3.7-3.5 3.7-9.4 0-12.8l-57.7-54.1c-3.5-3.3-9.1-3.2-12.4.4l-17.4 18.6c-3.3 3.5-3.1 9.1.4 12.4z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        issue-28/â€‹.eslintrc.json
      </div>
    </li>
    
  </menu><menu
    class="config"
  >
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="lg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-file-code fa-w-12 fa-fw fa-lg"
            data-icon="file-code"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M149.9 349.1l-.2-.2-32.8-28.9 32.8-28.9c3.6-3.2 4-8.8.8-12.4l-.2-.2-17.4-18.6c-3.4-3.6-9-3.7-12.4-.4l-57.7 54.1c-3.7 3.5-3.7 9.4 0 12.8l57.7 54.1c1.6 1.5 3.8 2.4 6 2.4 2.4 0 4.8-1 6.4-2.8l17.4-18.6c3.3-3.5 3.1-9.1-.4-12.4zm220-251.2L286 14C277 5 264.8-.1 252.1-.1H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48V131.9c0-12.7-5.1-25-14.1-34zM256 51.9l76.1 76.1H256zM336 464H48V48h160v104c0 13.3 10.7 24 24 24h104zM209.6 214c-4.7-1.4-9.5 1.3-10.9 6L144 408.1c-1.4 4.7 1.3 9.6 6 10.9l24.4 7.1c4.7 1.4 9.6-1.4 10.9-6L240 231.9c1.4-4.7-1.3-9.6-6-10.9zm24.5 76.9l.2.2 32.8 28.9-32.8 28.9c-3.6 3.2-4 8.8-.8 12.4l.2.2 17.4 18.6c3.3 3.5 8.9 3.7 12.4.4l57.7-54.1c3.7-3.5 3.7-9.4 0-12.8l-57.7-54.1c-3.5-3.3-9.1-3.2-12.4.4l-17.4 18.6c-3.3 3.5-3.1 9.1.4 12.4z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        el-file/â€‹.eslintrc.json
      </div>
    </li>
    
  </menu><menu
    class="config"
  >
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="lg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-file-code fa-w-12 fa-fw fa-lg"
            data-icon="file-code"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M149.9 349.1l-.2-.2-32.8-28.9 32.8-28.9c3.6-3.2 4-8.8.8-12.4l-.2-.2-17.4-18.6c-3.4-3.6-9-3.7-12.4-.4l-57.7 54.1c-3.7 3.5-3.7 9.4 0 12.8l57.7 54.1c1.6 1.5 3.8 2.4 6 2.4 2.4 0 4.8-1 6.4-2.8l17.4-18.6c3.3-3.5 3.1-9.1-.4-12.4zm220-251.2L286 14C277 5 264.8-.1 252.1-.1H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48V131.9c0-12.7-5.1-25-14.1-34zM256 51.9l76.1 76.1H256zM336 464H48V48h160v104c0 13.3 10.7 24 24 24h104zM209.6 214c-4.7-1.4-9.5 1.3-10.9 6L144 408.1c-1.4 4.7 1.3 9.6 6 10.9l24.4 7.1c4.7 1.4 9.6-1.4 10.9-6L240 231.9c1.4-4.7-1.3-9.6-6-10.9zm24.5 76.9l.2.2 32.8 28.9-32.8 28.9c-3.6 3.2-4 8.8-.8 12.4l.2.2 17.4 18.6c3.3 3.5 8.9 3.7 12.4.4l57.7-54.1c3.7-3.5 3.7-9.4 0-12.8l-57.7-54.1c-3.5-3.3-9.1-3.2-12.4.4l-17.4 18.6c-3.3 3.5-3.1 9.1.4 12.4z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        lintel/â€‹empty.json
      </div>
    </li>
    
  </menu><menu
    class="config"
  >
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="lg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-file-code fa-w-12 fa-fw fa-lg"
            data-icon="file-code"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M149.9 349.1l-.2-.2-32.8-28.9 32.8-28.9c3.6-3.2 4-8.8.8-12.4l-.2-.2-17.4-18.6c-3.4-3.6-9-3.7-12.4-.4l-57.7 54.1c-3.7 3.5-3.7 9.4 0 12.8l57.7 54.1c1.6 1.5 3.8 2.4 6 2.4 2.4 0 4.8-1 6.4-2.8l17.4-18.6c3.3-3.5 3.1-9.1-.4-12.4zm220-251.2L286 14C277 5 264.8-.1 252.1-.1H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48V131.9c0-12.7-5.1-25-14.1-34zM256 51.9l76.1 76.1H256zM336 464H48V48h160v104c0 13.3 10.7 24 24 24h104zM209.6 214c-4.7-1.4-9.5 1.3-10.9 6L144 408.1c-1.4 4.7 1.3 9.6 6 10.9l24.4 7.1c4.7 1.4 9.6-1.4 10.9-6L240 231.9c1.4-4.7-1.3-9.6-6-10.9zm24.5 76.9l.2.2 32.8 28.9-32.8 28.9c-3.6 3.2-4 8.8-.8 12.4l.2.2 17.4 18.6c3.3 3.5 8.9 3.7 12.4.4l57.7-54.1c3.7-3.5 3.7-9.4 0-12.8l-57.7-54.1c-3.5-3.3-9.1-3.2-12.4.4l-17.4 18.6c-3.3 3.5-3.1 9.1.4 12.4z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        lintel/â€‹invalid.json
      </div>
    </li>
    
  </menu><menu
    class="config"
  >
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="lg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-file-code fa-w-12 fa-fw fa-lg"
            data-icon="file-code"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M149.9 349.1l-.2-.2-32.8-28.9 32.8-28.9c3.6-3.2 4-8.8.8-12.4l-.2-.2-17.4-18.6c-3.4-3.6-9-3.7-12.4-.4l-57.7 54.1c-3.7 3.5-3.7 9.4 0 12.8l57.7 54.1c1.6 1.5 3.8 2.4 6 2.4 2.4 0 4.8-1 6.4-2.8l17.4-18.6c3.3-3.5 3.1-9.1-.4-12.4zm220-251.2L286 14C277 5 264.8-.1 252.1-.1H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48V131.9c0-12.7-5.1-25-14.1-34zM256 51.9l76.1 76.1H256zM336 464H48V48h160v104c0 13.3 10.7 24 24 24h104zM209.6 214c-4.7-1.4-9.5 1.3-10.9 6L144 408.1c-1.4 4.7 1.3 9.6 6 10.9l24.4 7.1c4.7 1.4 9.6-1.4 10.9-6L240 231.9c1.4-4.7-1.3-9.6-6-10.9zm24.5 76.9l.2.2 32.8 28.9-32.8 28.9c-3.6 3.2-4 8.8-.8 12.4l.2.2 17.4 18.6c3.3 3.5 8.9 3.7 12.4.4l57.7-54.1c3.7-3.5 3.7-9.4 0-12.8l-57.7-54.1c-3.5-3.3-9.1-3.2-12.4.4l-17.4 18.6c-3.3 3.5-3.1 9.1.4 12.4z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        lintel/â€‹common-js.cjs
      </div>
    </li>
    
  </menu><menu
    class="config"
  >
    <li
      class="item"
    >
      <div>
        <fa-icon
          class="ng-fa-icon"
          size="lg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-file-code fa-w-12 fa-fw fa-lg"
            data-icon="file-code"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M149.9 349.1l-.2-.2-32.8-28.9 32.8-28.9c3.6-3.2 4-8.8.8-12.4l-.2-.2-17.4-18.6c-3.4-3.6-9-3.7-12.4-.4l-57.7 54.1c-3.7 3.5-3.7 9.4 0 12.8l57.7 54.1c1.6 1.5 3.8 2.4 6 2.4 2.4 0 4.8-1 6.4-2.8l17.4-18.6c3.3-3.5 3.1-9.1-.4-12.4zm220-251.2L286 14C277 5 264.8-.1 252.1-.1H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48V131.9c0-12.7-5.1-25-14.1-34zM256 51.9l76.1 76.1H256zM336 464H48V48h160v104c0 13.3 10.7 24 24 24h104zM209.6 214c-4.7-1.4-9.5 1.3-10.9 6L144 408.1c-1.4 4.7 1.3 9.6 6 10.9l24.4 7.1c4.7 1.4 9.6-1.4 10.9-6L240 231.9c1.4-4.7-1.3-9.6-6-10.9zm24.5 76.9l.2.2 32.8 28.9-32.8 28.9c-3.6 3.2-4 8.8-.8 12.4l.2.2 17.4 18.6c3.3 3.5 8.9 3.7 12.4.4l57.7-54.1c3.7-3.5 3.7-9.4 0-12.8l-57.7-54.1c-3.5-3.3-9.1-3.2-12.4.4l-17.4 18.6c-3.3 3.5-3.1 9.1.4 12.4z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
      <div>
        lintel/â€‹src/â€‹app/â€‹.eslintrc.yaml
      </div>
    </li>
    
  </menu>
</lintel-configs>
`;
